src/transformers/trainer.py

```diff
@@ -3190,6 +3192,9 @@ def prediction_step(
             logits and labels (each being optional).
         """
         has_labels = False if len(self.label_names) == 0 else all(inputs.get(k) is not None for k in self.label_names)
+        # For CLIP-like models capable of returning loss values.
+        can_compute_loss = True if len(self.label_names) == 0 and self.can_return_loss else False
```

Comment:
We need to restrict to `len(self.label_names) == 0`.

For models that has `len(self.label_names) > 0`, we should check if the inputs contain the required labels by the model - which is done one line above `has_labels`.

Reply:
`can_compute_loss` is actually `can_compute_loss_without_labels`, but maybe a too long name.

Reply:
Can we rename all `can_compute_loss` to `loss_without_labels` then? It's more informative even if not completely perfect.